src/BreakpointBrowser/BreakpointBrowser.class.st

```diff
@@ -0,0 +1,83 @@
+"
+My class method ""open"" opens a Calypso method browser showing all the methods of the image that have halts or breakpoints in them. The different flavours of halts and breakpoints are also covered (break once, break on condition, haltIf: haltOnCount:, haltOnce...).
+
+By default, I do not show methods from my own package, nor methods registered on my ""blacklist"" class method. Blacklisted methods are meant to be methods from the core of Pharo that include halts for legitimate reasons (testing for example) and would probably pollute the experience of users if they were shown in the BreakpointBrowser.
+If you want to see all methods containing halts and breakpoints without any filtering, you can call my ""openUnfiltered"" class method. 
+"
+Class {
+	#name : #BreakpointBrowser,
+	#superclass : #Object,
+	#category : #BreakpointBrowser
+}
+
+{ #category : #filtering }
+BreakpointBrowser class >> blacklist [
+	"These methods will not be shown by BreakpointBrowser by default. They contain halts but are meant to contain them (for example, they can be testing that executing self halt does indeed raise a halt)
+	Format: Array of {class name symbol. method name symbol} arrays"
+	^ { 
+	{#ClyClassWithHalts. #methodWithHalts2}.
+	{#ClyClassWithHalts. #methodWithHalts1}.
+	{#ClyClassWithProblemMethods. #methodWithHalt}.
+	{#ClyClassWithProblemMethods. #methodWithHalt2}.
+	{#GTSUnitExampleFailingTest. #testWithHalt}.
+	{#Object. #openDebugger}.
+	{#Object. #openDebugger:}.
+	{#ObjectTest. #b}.
+	{#ObjectTest. #b1}.
+	{#ObjectTest. #testHaltIf}.
+	{#ObjectTest. #testHaltOnce}.
+	{#ObjectTest. #testHaltOnCount}.
+	{#UndefinedObjectTest. #testHaltIfNil}.
+	{#UndefinedObjectTest. #testIfNil}.
+	{#UndefinedObjectTest. #testIfNilIfNotNil}.
+	{#UndefinedObjectTest. #testIfNotNilIfNil}.
+	{#CodeSimulationTests. #methodWithHalt}.
+	{#RBRefactoryTestDataApp. #cruft}.
+	{#RBSmalllintTestObject. #codeCruftLeftInMethods}.
+	{#RBDummyRefactoryTestDataApp. #cruft}.
```

Comment:
Did you consider to mark those methods with special pragma instead of hardcoding such list?
Like add <testingHalt> into method which should not be shown as method with real halt.